### ISSUE TITLE
Support *.tgz archive extension in torchvision.datasets.utils.extract_archive

### ISSUE THREAD

**Issue by vadimkantorov** [pytorch/vision#1648]
I was updating my old code for CUB2011 dataset and found that now torchvision breaks when extracting a tgz file: https://github.com/vadimkantorov/metriclearningbench/blob/master/cub2011.py#L11

```
    download_and_extract_archive(self.url, self.root, filename=self.filename, md5=self.tgz_md5)
  File "/miniconda/lib/python3.7/site-packages/torchvision/datasets/utils.py", line 252, in download_and_extract_archive
    extract_archive(archive, extract_root, remove_finished)
  File "/miniconda/lib/python3.7/site-packages/torchvision/datasets/utils.py", line 234, in extract_archive
    raise ValueError("Extraction of {} not supported".format(from_path))
ValueError: Extraction of data/CUB_200_2011.tgz not supported
1
```

**Comment by surgan12** [comment]
@vadimkantorov, .tar.gz and .tgz seem to be identical and we have that support already but anyhow for preventing the code from breaking, we could add this.

**Comment by vadimkantorov** [comment]
@fmassa Are there any guidelines on contributing dataset classes, e.g. are file hash values necessary for enclosing, if yes - how many?

**Comment by fmassa** [comment]
Fixed via #1650 

@vadimkantorov 
> Are there any guidelines on contributing dataset classes, e.g. are file hash values necessary for enclosing, if yes - how many?

I would recommend first opening an issue discussing the dataset that you are thinking about adding. General guidelines are: it should have tests (with dummy data), if it download data from the internet, it should perform md5 checks, and it should be a dataset which is widely used for research.
